pybop/models/base_model.py

```diff
@@ -45,14 +45,27 @@ class BaseModel:
 
     """
 
-    def __init__(self, name="Base Model", parameter_set=None):
+    def __init__(
+        self, name="Base Model", parameter_set=None, check_params: Callable = None
```

Comment:
Are we able to replace `self._check_params()` with the callable arg `check_params` if it is provided? I didn't fully check, but it seems like we could overwrite `self._check_params()` if this arg is provided. This would clean up the attributes and logic if possible.